platform/ui/src/contextProviders/ViewportGridProvider.tsx

```diff
@@ -47,12 +47,13 @@ export function ViewportGridProvider({ children, service }) {
         return { ...state, ...{ activeViewportIndex: action.payload } };
       }
       case 'SET_DISPLAYSET_FOR_VIEWPORT': {
-        const {
-          viewportIndex,
-          displaySetInstanceUIDs,
-          viewportOptions,
-          displaySetOptions,
-        } = action.payload;
+        const payload = action.payload;
```

Comment:
The previous code was buggy in that calling setDisplaySetsForViewport would over-ride the viewport options and display set options EVEN if they were supposed to remain the same.  Just moved the default values here so that if values aren't provided, the old values or a default is used, and if the values are provided, then the new values replace the old ones.